### PR TITLE
doc/radosgw: explain multisite dynamic sharding

### PR DESCRIPTION
**Pull request by zdover23**
Add a note to doc/radosgw/dynamicresharding.rst and a note to doc/radosgw/multisite.rst that explains that dynamic resharding is not supported in releases prior to Reef.

This commit is made in response to a request from Mathias Chapelain.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/radosgw/dynamicresharding.rst [hunk 1]
```diff
@@ -37,8 +37,8 @@ resharding tasks, one at a time.
 Multisite
 =========
 
-Dynamic resharding is not supported in a multisite environment.
-
+Prior to the Reef release, RADOSGW does not support dynamic resharding in a
```
Comment (anthonyeleven) [inline doc/radosgw/dynamicresharding.rst:40]: Sometimes we call it RGW, sometimes recently Ceph Object Gateway, but it shouldn't be all uppercase.

File: doc/radosgw/multisite.rst [hunk 2]
```diff
@@ -1582,14 +1582,23 @@ Supported Features
...
+Resharding
 ~~~~~~~~~~
 
-Allows buckets to be resharded in a multisite configuration without interrupting the replication of their objects. When ``rgw_dynamic_resharding`` is enabled, it runs on each zone independently, and zones may choose different shard counts for the same bucket. When buckets are resharded manually with ``radosgw-admin bucket reshard``, only that zone's bucket is modified. A zone feature should only be marked as supported after all of its radosgws and osds have upgraded.
+Allows buckets to be resharded in a multisite configuration without
+interrupting the replication of their objects. When ``rgw_dynamic_resharding``
+is enabled, it runs on each zone independently, and zones may choose different
+shard counts for the same bucket. When buckets are resharded manually with
+``radosgw-admin bucket reshard``, only that zone's bucket is modified. A zone
+feature should only be marked as supported after all of its radosgws and osds
```
Comment (anthonyeleven) [inline doc/radosgw/multisite.rst:1593]: OSDs
same radosgws as above, shouldn't be all lowercase either.

**Review by anthonyeleven — APPROVED** [review]
Capitalization requests

File: doc/radosgw/dynamicresharding.rst (hunk 1 above)
Comment (zdover23) [reply doc/radosgw/dynamicresharding.rst]: Good looking out.

File: doc/radosgw/multisite.rst (hunk 2 above)
Comment (zdover23) [reply doc/radosgw/multisite.rst]: ibid

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/51585 - Reef backport
https://github.com/ceph/ceph/pull/51586 - Quincy backport
https://github.com/ceph/ceph/pull/51587 - Pacific backport

**Pull request merged**